src/rlgl.h

```diff
@@ -474,18 +474,6 @@ typedef enum {
     RL_PIXELFORMAT_COMPRESSED_ASTC_8x8_RGBA        // 2 bpp
 } rlPixelFormat;
 
-// Texture parameters: filter mode
-// NOTE 1: Filtering considers mipmaps if available in the texture
-// NOTE 2: Filter is accordingly set for minification and magnification
-typedef enum {
-    RL_TEXTURE_FILTER_POINT = 0,        // No filter, just pixel approximation
-    RL_TEXTURE_FILTER_BILINEAR,         // Linear filtering
-    RL_TEXTURE_FILTER_TRILINEAR,        // Trilinear filtering (linear with mipmaps)
-    RL_TEXTURE_FILTER_ANISOTROPIC_4X,   // Anisotropic filtering 4x
-    RL_TEXTURE_FILTER_ANISOTROPIC_8X,   // Anisotropic filtering 8x
-    RL_TEXTURE_FILTER_ANISOTROPIC_16X,  // Anisotropic filtering 16x
-} rlTextureFilter;
-
```

Comment:
Please don't remove the `rlTextureFilter`.
Although it's not used on `raylib`, it's used when using `rlgl` standalone.

Reply:
I committed the changes locally, but before pushing, could you let me know to what extent it is being used? I searched thoroughly but couldn't find any usage in `rlgl`

Reply:
When using `rlgl` standalone (where raylib is not present, so no `raylib.h`, no `rtextures`, etc) it's used with (to set the correct) `rlTextureParameters()` similarly to `SetTextureFilter()` ([ref](https://github.com/raysan5/raylib/blob/602583bcdbc46f317915c811a7e8e2416e65bbc7/src/rtextures.c#L4360)).

Reply:
Sincerely, I don't know... I usually like to simplify code and function parameters... but I agree that `SetTextureFilter()` uses them and it could be a reference for `rlgl` users.

About the `format` parameter in `rlGenTextureMipmaps()`, it was kept there for a future implementation for OpenGL 1.1 that required the data size in memory... but implementation was removed at some point in raylib live...

Reply:
I'd like to point a parallel with Blend Mode.
The [BlendMode](https://github.com/raysan5/raylib/blob/83f6f3dabeb14b8f5268bf3abdcc898f50cfeb17/src/raylib.h#L902-L911) enum mirrors [rlBlendMode](https://github.com/raysan5/raylib/blob/83f6f3dabeb14b8f5268bf3abdcc898f50cfeb17/src/rlgl.h#L493-L502C3).
All [BeginBlendMode()](https://github.com/raysan5/raylib/blob/83f6f3dabeb14b8f5268bf3abdcc898f50cfeb17/src/rcore.c#L1144-L1147) / [EndBlendMode()](https://github.com/raysan5/raylib/blob/83f6f3dabeb14b8f5268bf3abdcc898f50cfeb17/src/rcore.c#L1150-L1153) do is call [rlSetBlendMode()](https://github.com/raysan5/raylib/blob/83f6f3dabeb14b8f5268bf3abdcc898f50cfeb17/src/rlgl.h#L2101-L2136).
And [rlSetBlendMode()](https://github.com/raysan5/raylib/blob/83f6f3dabeb14b8f5268bf3abdcc898f50cfeb17/src/rlgl.h#L2108-L2130) handles the switch from there.

The Texture Filter is basically the same.
The [TextureFilter](https://github.com/raysan5/raylib/blob/83f6f3dabeb14b8f5268bf3abdcc898f50cfeb17/src/raylib.h#L868-L875) enum mirrors [rlTextureFilter](https://github.com/raysan5/raylib/blob/83f6f3dabeb14b8f5268bf3abdcc898f50cfeb17/src/rlgl.h#L483-L490).
All [SetTextureFilter()](https://github.com/raysan5/raylib/blob/83f6f3dabeb14b8f5268bf3abdcc898f50cfeb17/src/rtextures.c#L4360-L4423) does is calls to [rlTextureParameters()](https://github.com/raysan5/raylib/blob/83f6f3dabeb14b8f5268bf3abdcc898f50cfeb17/src/rtextures.c#L4369-L4420).
But it was chosen to handle the switch at [SetTextureFilter()](https://github.com/raysan5/raylib/blob/83f6f3dabeb14b8f5268bf3abdcc898f50cfeb17/src/rtextures.c#L4362-L4422) instead of on `rlgl` (thus, incidentally, not using `rlTextureFilter`).

It would be nice to keep `rlTextureFilter` at least for the sake of structural consistency since Texture Filtering could very well be implemented inside `rlgl`, similarly to [rlSetBlendMode()](https://github.com/raysan5/raylib/blob/83f6f3dabeb14b8f5268bf3abdcc898f50cfeb17/src/rlgl.h#L2101-L2136).

Reply:
Alright, understood, I just canceled the change

Reply:
Thank you.